app/src/main/java/de/ur/explure/viewmodel/WordSearchViewModel.kt

```diff
@@ -3,19 +3,116 @@ package de.ur.explure.viewmodel
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.algolia.search.client.ClientSearch
+import com.algolia.search.model.APIKey
+import com.algolia.search.model.ApplicationID
+import com.algolia.search.model.IndexName
+import com.algolia.search.model.search.Query
 import de.ur.explure.model.route.Route
 import de.ur.explure.repository.route.RouteRepositoryImpl
 import de.ur.explure.utils.FirebaseResult
 import kotlinx.coroutines.launch
+import timber.log.Timber
 
+@Suppress("UnnecessaryParentheses")
 class WordSearchViewModel(
     private val routeRepo: RouteRepositoryImpl
 ) : ViewModel() {
 
     var searchedRoutes: MutableLiveData<List<Route>> = MutableLiveData()
+    var noRoutes = MutableLiveData<Boolean>(false)
 
     fun getSearchedRoutes(message: String) {
         viewModelScope.launch {
+
+            val applicationID = ApplicationID("CRDAJVEWKR")
```

Comment:
Das hier sollte eig alles in einen eigenen Algolia Provider, aber das können wir auch später exportieren. Und ApiKey sollten eig nicht einfach so im Code gehardcodet werden.

Reply:
der APIKey ist der only-read Key, den man zumindest offen verwenden kann "This is the public API key to use in your frontend code"